test/test_grid_search.py

```diff
@@ -0,0 +1,163 @@
+        feat3 = [int(x) for x in '1111111' '1111111' '111111']
+        X = pd.DataFrame({"feat1": feat1, "feat2": feat2, "feat3": feat3})
+
+        target = gs.BinaryClassificationGridSearch()
+        with pytest.raises(RuntimeError, match=message):
+            _ = target.grid_search_binary_protected_attribute(LeastSquaresLearner(), X, Y, A, None, None)
+        with pytest.raises(RuntimeError, match=message):
+            _ = target.grid_search_binary_protected_attribute(LeastSquaresLearner(), X, Y, A, [-1,0,1], 3)
+
+    def test_grid_bad_Y_shape(self):
+        message = r"Supplied y not 1d vector"
+
+        A = pd.Series([int(x) for x in '0000000' '1111111' '010101'])
+        feat1 = [int(x) for x in '0110101' '0111101' '001011']
+        feat2 = [int(x) for x in '0000100' '0000011' '111111']
+        feat3 = [int(x) for x in '1111111' '1111111' '111111']
+        X = pd.DataFrame({"feat1": feat1, "feat2": feat2, "feat3": feat3})
+
+        target = gs.BinaryClassificationGridSearch()
+        badY = np.ones((2,3))
+        with pytest.raises(RuntimeError, match=message):
+            _ = target.grid_search_binary_protected_attribute(LeastSquaresLearner(), X, badY, A)
+
+    def test_grid_bad_A_shape(self):
+        message = r"Supplied protected_attribute not 1d vector"
+
+        Y = pd.Series([int(x) for x in '0110100' '0010111' '001111'])
+        feat1 = [int(x) for x in '0110101' '0111101' '001011']
+        feat2 = [int(x) for x in '0000100' '0000011' '111111']
+        feat3 = [int(x) for x in '1111111' '1111111' '111111']
+        X = pd.DataFrame({"feat1": feat1, "feat2": feat2, "feat3": feat3})
+
+        target = gs.BinaryClassificationGridSearch()
+        badA = np.ones((2,5))
+        with pytest.raises(RuntimeError, match=message):
+            _ = target.grid_search_binary_protected_attribute(LeastSquaresLearner(), X, Y, badA)
+
+    def test_grid_A_Y_mismatch(self):
+        message = r"Supplied protected_attribute and y not same length"
+
+        A = pd.Series([int(x) for x in '0000000' '1111111' '010101' '1'])
+        Y = pd.Series([int(x) for x in '0110100' '0010111' '001111'])
+        feat1 = [int(x) for x in '0110101' '0111101' '001011']
+        feat2 = [int(x) for x in '0000100' '0000011' '111111']
+        feat3 = [int(x) for x in '1111111' '1111111' '111111']
+        X = pd.DataFrame({"feat1": feat1, "feat2": feat2, "feat3": feat3})
+
+        target = gs.BinaryClassificationGridSearch()
+        with pytest.raises(RuntimeError, match=message):
+            _ = target.grid_search_binary_protected_attribute(LeastSquaresLearner(), X, Y, A)
+
+    def test_grid_X_Y_mismatch(self):
+        message = r"Supplied x and y do not have same number of rows"
+
+        A = pd.Series([int(x) for x in '0000000' '1111111' '010101' '1'])
+        Y = pd.Series([int(x) for x in '0110100' '0010111' '001111' '0'])
+        feat1 = [int(x) for x in '0110101' '0111101' '001011']
```

Comment:
looks like we're repeating these. Do you mind putting them as constants at the top of the file?

Reply:
Refactored